File: 899-binary-gap/binary-gap.cpp
class Solution {
public:
    int binaryGap(int n) {
        
        int curr = -1;
        int maxi = 0;
        for(int idx = 0; idx < 32 && n > 0; idx++) {

            if((1 & n) == 1) {
                cout<< n << endl;
                curr++;
                maxi = max(maxi, curr);
                curr = 0;
            }
            else if(curr != -1) curr++;


            n = n >> 1;
            
        }


        return max(maxi, curr);
    }
};

File: 899-binary-gap/binary-gap.js
/**
 * @param {number} n
 * @return {number}
 */
var binaryGap = function(n) {
    
    let str = n.toString(2);

    let start = str.indexOf('1');

    if(start === -1) return 0;

    let max = 0;

    for(let i = start + 1; i < str.length; i++) {
        if(str[i] === '1') {
            max = Math.max(max, i - start);
            start = i;
        }
    }
    return max;
};